linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/restful/ConfigurationTemplateRestfulApi.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.linkis.basedatamanager.server.restful;
+
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import org.apache.linkis.basedatamanager.server.domain.ConfigurationConfigKey;
+import org.apache.linkis.basedatamanager.server.request.ConfigurationTemplateSaveRequest;
+import org.apache.linkis.basedatamanager.server.response.EngineLabelResponse;
+import org.apache.linkis.basedatamanager.server.service.ConfigurationTemplateService;
+import org.apache.linkis.server.Message;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.InvalidParameterException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This module is designed to manage configuration parameter templates
+ */
+@RestController
+@RequestMapping(path = "/basedata-manager/configuration-template")
+public class ConfigurationTemplateRestfulApi {
+
+    @Autowired
+    ConfigurationTemplateService configurationTemplateService;
+
+    @ApiOperation(value = "save", notes = "save a configuration template", httpMethod = "POST")
+    @RequestMapping(path = "/save", method = RequestMethod.POST)
+    public Message add(@RequestBody ConfigurationTemplateSaveRequest request) {
+        if (Objects.isNull(request) || StringUtils.isEmpty(request.getEngineLabelId()) || StringUtils.isEmpty(request.getKey()) ||
```

Comment:
need add audit log for restful api such as:
```
  String userName=ModuleUserUtils.getOperationUser(request, "Save a configuration template, engineLabelId :" + request.getEngineLabelId());
```
Only administrators have permissions, and role judgments need to be added